From 02148814fe284c03c0158a68bdaa9deac9d084ce Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 13 Dec 2022 16:14:34 +0200
Subject: [PATCH] [SW-66344] configs: update to buildroot 2022.08

As part of update to buildroot 2022.08, should use
disable the safety related configs which were enabled
by default in the updated buildroot.

This is related to the following patch in buildroot
2021-05-03 - Config.in: enable FORTIFY_SOURCE, PIC/PIE,
RELRO, SSP by default <Fabrice Fontaine>

Change-Id: Icdb4758d04794b5147e88d7b63860659fad416f3
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/configs/gaudi2_arc_defconfig b/configs/gaudi2_arc_defconfig
index bab18a1..58a7e33 100644
--- a/configs/gaudi2_arc_defconfig
+++ b/configs/gaudi2_arc_defconfig
@@ -21,3 +21,7 @@
 BR2_PACKAGE_ZEPHYR_ARC_PID=y
 BR2_PACKAGE_ZEPHYR_ARC_MGMT=y
 BR2_PACKAGE_ZEPHYR_ARC_PREBOOT=y
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
\ No newline at end of file
diff --git a/configs/gaudi2_arm_defconfig b/configs/gaudi2_arm_defconfig
index 9f1f72a..9f53f36 100644
--- a/configs/gaudi2_arm_defconfig
+++ b/configs/gaudi2_arm_defconfig
@@ -68,3 +68,7 @@
 BR2_PACKAGE_ZEPHYR_ARM_AGENT_SERDES=y
 BR2_PACKAGE_ZEPHYR_ARM_AGENT_HBM=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
\ No newline at end of file
diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index 65edc0a..aee005b 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -29,3 +29,7 @@
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
 BR2_POST_IMAGE_FW_LOADER_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/agent-fw_loader.its"
 BR2_POST_IMAGE_AGENTS_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/agent.its"
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
diff --git a/configs/gaudi3_arm_defconfig b/configs/gaudi3_arm_defconfig
index eb8dd14..f5b47cd 100644
--- a/configs/gaudi3_arm_defconfig
+++ b/configs/gaudi3_arm_defconfig
@@ -10,3 +10,7 @@
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
diff --git a/configs/gaudi_defconfig b/configs/gaudi_defconfig
index b7d5f31..5e06567 100644
--- a/configs/gaudi_defconfig
+++ b/configs/gaudi_defconfig
@@ -62,3 +62,7 @@
 BR2_PACKAGE_BTL=y
 BR2_PACKAGE_EEPROM_UTIL=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
\ No newline at end of file
diff --git a/configs/greco_arc_defconfig b/configs/greco_arc_defconfig
index 406bedf..3f528a4 100644
--- a/configs/greco_arc_defconfig
+++ b/configs/greco_arc_defconfig
@@ -20,3 +20,7 @@
 BR2_PACKAGE_ZEPHYR_ARC_PID=y
 BR2_PACKAGE_ZEPHYR_ARC_MGMT=y
 BR2_PACKAGE_HOST_ARCBOOT=y
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
\ No newline at end of file
diff --git a/configs/greco_arc_elf32_defconfig b/configs/greco_arc_elf32_defconfig
index bbbc115..3595a61 100644
--- a/configs/greco_arc_elf32_defconfig
+++ b/configs/greco_arc_elf32_defconfig
@@ -21,3 +21,7 @@
 # BR2_TARGET_ROOTFS_TAR is not set
 BR2_PACKAGE_PREBOOT=y
 BR2_PACKAGE_ARCBOOT=y
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
\ No newline at end of file
diff --git a/configs/greco_arm_defconfig b/configs/greco_arm_defconfig
index 6783945..c52bd90 100644
--- a/configs/greco_arm_defconfig
+++ b/configs/greco_arm_defconfig
@@ -63,3 +63,7 @@
 BR2_PACKAGE_BTL=y
 BR2_PACKAGE_EEPROM_UTIL=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
\ No newline at end of file
